src/app/shared/components/fss-footer/fss-footer.component.spec.ts

```diff
@@ -1,42 +1,71 @@
-    await TestBed.configureTestingModule({
-      imports: [
-        RouterTestingModule,        
-      ],
-      declarations: [ FssFooterComponent ]
-    })
-    .compileComponents();
-  });
-
-  beforeEach(() => {
-    fixture = TestBed.createComponent(FssFooterComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  });
-
-  it('should create', () => {
-    expect(component).toBeTruthy();
-  });
-
-  it('should have footer component', () => {    
-    let compiled = fixture.nativeElement;
-    let footer = compiled.querySelector('ukho-footer');
-    expect(footer).not.toBeNull();    
-  });
   
-  it('should have 2 navigation items', () => {  
-    const app = fixture.componentInstance;
-    expect(app.navigation.length).toEqual(2);
-    expect(app.navigation[0].title).toEqual("Privacy policy");
-    expect(app.navigation[1].title).toEqual("Accessibility");
-  });
+    beforeEach(async () => {
+      await TestBed.configureTestingModule({
+        imports: [RouterTestingModule],
+        declarations: [FssFooterComponent , FooterComponent],
+        schemas: [NO_ERRORS_SCHEMA]
+      }).compileComponents();
+    });
+
+
+      it('should have footer component', () => {    
+        const fixture = TestBed.createComponent(FssFooterComponent);
+        const footer = fixture.debugElement.nativeElement.querySelector('ukho-footer');
+        const header = fixture.debugElement.nativeElement.querySelector('ukho-header');
+        expect(footer).not.toBeNull();
+        expect(header).toBeFalsy();    
+      });
+
+      test('should render image and navigation in ukho-footer', () =>{
+        const fixture = TestBed.createComponent(FssFooterComponent);
+        fixture.detectChanges();     
+        const footerAnchorTags = fixture.debugElement.queryAll(By.css('a'));
+        for(var i=0; i < footerAnchorTags.length; i++){
+          var img = footerAnchorTags[i];
+          expect(img.attributes['href']).not.toBe(null);
+        }
+      });
+
+      test('should render the text element of ukho-footer', () =>{
+      const fixture = TestBed.createComponent(FssFooterComponent);
+      fixture.detectChanges();     
+      const footer = fixture.debugElement.query(By.css('ukho-footer')).nativeElement;
+      expect(footer.querySelector('p').textContent).not.toBeNull();
+      expect(footer.querySelector('p').textContent).toContain('@ Crown copyright 2021 UK Hydrographic office');  
+      expect(footer.querySelector('p').textContent).not.toContain('File Share Service');  
+      });
+
+     
+      test('should exist', () => {
+        component = new FssFooterComponent();
+        component.ngOnInit();
+        expect(component).toBeDefined();
+      });
+              
+      test('should exist copyright statement in footer', () => {
+        component = new FssFooterComponent();
+        component.ngOnInit();
+        expect(component.text).toEqual('@ Crown copyright 2021 UK Hydrographic office');
```

Comment:
Can these values be changed to constants so they are easier to change in the future?

Reply:
created separate file for such static values